diff -ur /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/ext/gvpr/dot2ruby.g /home/ryoqun/ruby-graphviz-1.0.9/lib/ext/gvpr/dot2ruby.g
--- /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/ext/gvpr/dot2ruby.g	2013-10-10 20:33:38.033918001 +0900
+++ /home/ryoqun/ruby-graphviz-1.0.9/lib/ext/gvpr/dot2ruby.g	2013-10-10 20:31:02.434889000 +0900
@@ -45,6 +45,7 @@
   string stringify( string s ) {
     string sout;
     sout = gsub(s, "\"", "\\\"");
+    sout = gsub(sout, "\'", "\\\'");
     sout = gsub(sout, "@", "\\@");
     sout = gsub(sout, "$", "\\$");
     return( sout );
@@ -72,7 +73,7 @@
   while( attr != "" ) {
     attrv = aget( $, attr );
     if( attrv != "" ) {
-      printf( "  graph_%s[:%s] = '%s'\n", rubyfy($.name), attr, attrv );
+      printf( "  graph_%s[:%s] = '%s'\n", rubyfy($.name), attr, stringify(attrv) );
     }
     attr = nxtAttr( $, "G", attr );
   }
diff -ur /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz/dot2ruby.rb /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz/dot2ruby.rb
--- /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz/dot2ruby.rb	2013-10-10 20:33:38.033918001 +0900
+++ /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz/dot2ruby.rb	2013-10-10 20:21:40.430889000 +0900
@@ -15,6 +15,7 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 
+require 'shellwords'
 require 'graphviz/ext'
 require 'graphviz/utils'
 
@@ -30,7 +31,7 @@
   end
 
   def run( xFile ) #:nodoc:
-    xCmd = "#{@xGvprPath} -f #{@gvprScript} -a #{@xOutFormat} #{xFile}"
+    xCmd = "#{@xGvprPath} -f #{Shellwords.escape @gvprScript} -a #{@xOutFormat} #{Shellwords.escape xFile}"
     xOutput = output_from_command( xCmd )
     if @xOutFile.nil?
       puts xOutput
@@ -42,8 +43,9 @@
   end
 
   def eval( xFile ) #:nodoc:
-    xCmd = "#{@xGvprPath} -f #{@gvprScript} -a \"-\" #{xFile}"
+    xCmd = "#{@xGvprPath} -f #{Shellwords.escape @gvprScript} -a \"-\" #{Shellwords.escape xFile}"
     xOutput = output_from_command( xCmd )
+    p xOutput
     instance_eval(xOutput)
     return @_graph_eval
   end
diff -ur /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz/types/esc_string.rb /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz/types/esc_string.rb
--- /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz/types/esc_string.rb	2013-10-10 20:33:38.037918001 +0900
+++ /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz/types/esc_string.rb	2013-10-10 08:27:09.598860000 +0900
@@ -6,7 +6,7 @@
       end
 
       def output
-        return @data.to_s.inspect.gsub( "\\\\", "\\" )
+        return @data.to_s.inspect.gsub("\\\\", "\\")
       end
 
       alias :to_gv :output
diff -ur /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz.rb /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz.rb
--- /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/lib/graphviz.rb	2013-10-10 20:33:38.033918001 +0900
+++ /home/ryoqun/ruby-graphviz-1.0.9/lib/graphviz.rb	2013-10-10 08:29:04.418860000 +0900
@@ -586,6 +586,7 @@
           xCmd = "\"#{cmd}\" -q#{@errors} #{xExternalLibraries} #{xOtherOptions} #{xOutputWithFile} #{xOutputWithoutFile} #{t.path}"
         end
 
+        p xCmd
         xOutput << output_from_command( xCmd )
       end
 
@@ -949,7 +950,7 @@
 
     if (options[:force] or str.match( /\A[a-zA-Z_]+[a-zA-Z0-9_]*\Z/ ).nil?)
       unless options[:unquote_empty] and str.size == 0
-        '"' + str.gsub('"', '\\"').gsub("\n", '\\\\n') + '"'
+        '"' + str.gsub("'", "\\'").gsub('"', '\\"').gsub("\n", '\\\\n') + '"'
       end
     else
       str
Only in /var/lib/gems/1.9.1/gems/ruby-graphviz-1.0.9/: .yardoc
